### PR TITLE
stringify cson with no indent

### LINKED ISSUE #36: stringify defaulting to JSON
Why does the stringify method default to JSON if the indentation level is falsey? 
This in unexpected and seems unnecessary as you could create valid minified cson

### PR DESCRIPTION
**Pull request by charlierudolph**
fixes #36

@jkrems @johan

### CODE REVIEW

File: src/stringify.coffee [hunk 1]
```diff
@@ -81,31 +79,42 @@ module.exports = (data, visitor, indent) ->
...
       key = JSON.stringify key unless key.match jsIdentifierRE
-      serializedValue = visitNode value
-      if isObject value
-        if serializedValue == ''
-          "#{ key }: {}"
+      serializedValue = visitNode value, bracesRequired: !indent
+      if indent
+        serializedValue = if isObject value
+          "\n#{ indentLines serializedValue }"
         else
```
Comment (jkrems) [inline src/stringify.coffee]: The indentation here doesn't seem to align

**Comment by jkrems** [comment]
Wow, that was quick! One small indentation thing but other than that LGTM.

File: src/stringify.coffee (hunk 1 above)
Comment (charlierudolph) [inline src/stringify.coffee]: Thats the proper place for the else. Its matching if is only two lines above, not three

Comment (johan) [inline src/stringify.coffee]: I think it does – lines 96 and 98 go together, not 95 and 98.

Comment (jkrems) [inline src/stringify.coffee]: Ah, got it. Sorry. Could you maybe change this to something like this:

```coffee
        serializedValue =
          if isObject value
            "\n#{ indentLines serializedValue }"
          else
            " #{ serializedValue }"
```

That way it's a bit easier to scan the left side of the code and grok the structure. Partially a question of personal taste but I'm fairly certain that matches our (informal) code style.

File: src/stringify.coffee [hunk 2]
```diff
@@ -81,31 +79,42 @@ module.exports = (data, visitor, indent) ->
...
   visitArray = (arr) ->
-    items = arr.map (value) ->
-      serializedValue = visitNode value
-      if isObject value
-        "{#{ newlineWrap indentLines serializedValue }}"
-      else
-        serializedValue
+    items = arr.map (value) -> visitNode value, bracesRequired: true
+
+    serializedItems = if indent
```
Comment (jkrems) [inline src/stringify.coffee]: Same here - I'd prefer if we keep `if` and `else` tokens on one height.

**Comment by charlierudolph** [comment]
@jkrems all comments addressed

**Comment by jkrems** [comment]
v1.1.0 - sorry for the delay, busy week.

**Comment by charlierudolph** [comment]
No worries. Glad to see this merged. Thanks!

**Comment by kevinsawicki** [comment]
So looks like this changed the following case:

```coffee
require('cson-parser').stringify({a:{}}, null, 2)
```

### Before

```coffee
a: {}
```

### After

```coffee
a:
  {}
```

Was this intentional?  It looks kind of weird to see the empty `{} indented and the behavior seems to differ from `JSON.stringify`

**Comment by jkrems** [comment]
@kevinsawicki Sorry for the regressions, fixed in v1.1.1 thanks to @charlierudolph.

**Pull request merged**